[kratos] Add Ory Kratos service

pvannierop opened the pull request:
**Description of the change**

This PR will add the Ory Kratos service. The Ory Kratos service will be the new solution for User and Identity management in RADAR-base.

keyvaann commented on helmfile.d/15-ory.yaml:
```diff
@@ -0,0 +1,85 @@
+bases:
```
Since Ory is needed for all installations let's add it to `10-base.yaml` instead of having its own separate file

pvannierop replied:
Can't the same be said about Management Portal? It is in its own file. As an alternative, we can rename 10-managementportal.yaml to 10-authentication.yaml or 10-identity-management.yaml and add both MP and Ory-Kratos. What do you think?

keyvaann replied:
In general I'd like to reduce number of helmfiles and merge them whenever it makes sense. I'm fine with changing the names, maybe we can merge both MP and Ory into 10-base.yaml.
If you won't want to put them into 10-base.yaml, since MP and appconfig aren't only about authentication or identity management I think we should have a more descriptive name.

Bdegraaf1234 replied:
IMO, put it in `10-base.yaml`.

app-config is the most optional but you can still disable it. I don't think there is a big benefit to it being in a separate helmfile

pvannierop replied:
Merged into `10-base.yaml`!

pvannierop replied:
Because postgres is difined in the 10-managementporta.yaml file, I moved kratos to this helmfile.

keyvaann commented on helmfile.d/15-ory.yaml:
```diff
@@ -0,0 +1,85 @@
+    chart: ory/kratos
+    version: {{ .Values.kratos._chart_version }}
+    installed: {{ .Values.kratos._install }}
+    timeout: {{ add .Values.base_timeout .Values.kratos._extra_timeout }}
```
Please also add <<: *logFailedRelease to the release, similar to other ones.

pvannierop replied:
Done!

keyvaann commented on etc/base.yaml:
```diff
@@ -208,6 +214,18 @@ app_config_frontend:
+# --------------------------------------------------------- 15-ory.yaml ---------------------------------------------------------
+
+kratos:
+  _install: false
```
Isn't Ory core part of the RB now?

pvannierop replied:
Indeed, I will activate it by default.

keyvaann commented on helmfile.d/15-ory.yaml:
```diff
@@ -0,0 +1,85 @@
+        value: postgres://{{ .Values.management_portal.postgres.user }}:{{ .Values.management_portal.postgres.password }}@{{ .Values.management_portal.postgres.host }}:{{ .Values.management_portal.postgres.port }}/kratos
+      - name: kratos.config.courier.smtp.connection_uri
+        #  Note: encoding of "/" in password is necessary for the smtp connection_uri because kratos is not able to handle this.
+        value: smtp://{{ .Values.management_portal.smtp.username }}:{{ replace "/" "%2F" .Values.management_portal.smtp.password }}@{{ .Values.management_portal.smtp.host }}:587
```
SMTP port should also be configurable

pvannierop replied:
Done!

keyvaann commented on helmfile.d/15-ory.yaml:
```diff
@@ -0,0 +1,85 @@
+      - name: serverName
+        value: {{ .Values.server_name }}
+      - name: kratos.config.dsn
+        value: postgres://{{ .Values.management_portal.postgres.user }}:{{ .Values.management_portal.postgres.password }}@{{ .Values.management_portal.postgres.host }}:{{ .Values.management_portal.postgres.port }}/kratos
```
It's better that the database name is also configurable

pvannierop replied:
Done!

keyvaann commented on etc/base.yaml:
```diff
@@ -182,8 +182,12 @@ management_portal:
   _chart_version: 1.1.4
   _extra_timeout: 210
   replicaCount: 1  # should be 1
+  # FIXME: Should the postgres database connection info not be a global property?
```
Having a FIXME in this file might be confusing for the users, I think it's better to create a github issue for it

pvannierop replied:
Issue [created](https://github.com/RADAR-base/RADAR-Kubernetes/issues/269) and comment removed!

keyvaann commented on etc/kratos_ui/values.yaml:
```diff
@@ -0,0 +1,34 @@
+    - secretName: radar-base-tls
+      hosts:
+        - localhost
+        # -- Set this to ORY Kratos's Admin URL
```
The indentation isn't correct

pvannierop replied:
Fixed!

keyvaann commented on helmfile.d/15-ory.yaml:
```diff
@@ -0,0 +1,85 @@
+
+releases:
+  - name: kratos
+    chart: ory/kratos
```
Please mirror the chart in `external` directory of radar-helm-charts and refer it from there similar to other external charts

pvannierop replied:
I copied the Ory helm charts the radar-helm-charts repo. The PR is [here](https://github.com/RADAR-base/radar-helm-charts/pull/202). I updated the chart references in _15-ory.html_ accordingly.

keyvaann replied:
Thanks, please also do a quick check if kratos can be successfully installed from our mirror

Bdegraaf1234 replied:
@mpgxvii Did you ever install kratos using the latest radar-kubernetes and helm chart?

pvannierop commented on etc/kratos/values.yaml:
```diff
@@ -0,0 +1,275 @@
+
+    courier:
+      smtp:
+        from_address: radar@thehyve.nl
```
But it is configurable, right?

keyvaann replied:
It is but I think it should be in `base.yaml` since it will be changed in every installation.

Bdegraaf1234 replied:
agreed

pvannierop replied:
Ok the most simple solution would be to write this entry in base.yaml like so:
```
kratos:
  _install: true
  _chart_version: 0.43.0
  _extra_timeout: 0
  ....
  kratos:
    courier:
      smtp:
        from_address: radar@thehyve.nl
```

However, since SMTP settings are also used by MP (and possibly other services), we should consider defining this as global properties and inject this in the `set:` section of helmfile where needed.

keyvaann commented:
There seems to be an issue with the Kratos migration
https://github.com/RADAR-base/RADAR-Kubernetes/actions/runs/9515630882/job/26230258308?pr=266#step:11:38

pvannierop commented:
@keyvaann Yes, the problem was that we moved the kratos definitions to the 10-base.yaml file, but the postgres dependency was only defined in 10-managementportal.yaml. I corrected this and now it installs fine.
